packages/datadog-plugin-apollo-gateway/src/fetch.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const TracingPlugin = require('../../dd-trace/src/plugins/tracing')
+const { storage } = require('../../datadog-core')
+
+class ApolloGatewayFetchPlugin extends TracingPlugin {
+  static get id () { return 'apollo-gateway' }
+  static get operation () { return 'fetch' }
+  static get type () { return 'apollo-gateway' }
+  static get kind () { return 'server' }
+
+  static get prefix () {
+    return 'tracing:apm:apollo-gateway:fetch'
+  }
+
+  bindStart (ctx) {
+    const store = storage.getStore()
+    const childOf = store ? store.span : null
+
+    const spanData = {
+      childOf,
+      service: this.config.service,
+      type: this.constructor.type,
+      meta: {}
+    }
+
+    if (ctx?.attributes?.service) { spanData.meta['serviceName'] = ctx?.attributes?.service }
```

Comment:
Storing `ctx?.attributes?.service` in a variable would avoid repeating that code.